src/custom/state/enhancedTransactions/hooks/index.ts

```diff
@@ -105,3 +118,26 @@ export function useTransactionsByHash({ hashes }: { hashes: string[] }): Enhance
     }, {})
   }, [allTxs, hashes])
 }
+
+export function useAllClaimingTransactions() {
+  const transactionsMap = useAllTransactions()
+  const transactions = Object.values(transactionsMap)
+
+  return useMemo(() => {
+    return transactions.filter((tx) => !!tx.claim)
+  }, [transactions])
+}
```

Comment:
This is the one from Uni I mentioned which is kinda similar to yours

https://github.com/gnosis/cowswap/blob/6fb3ded2ba3bbc284477b1c2d78bf2a4bf201538/src/state/transactions/hooks.tsx#L106

Reply:
ah, true. i like mine more and it's a bit different :)

Reply:
Of course, just keep in mind to update references to your hook / mod existing hook to not rely in the old logic